### ISSUE TITLE
Fable compilation stalls when types parametrized by units of measure are used

### ISSUE THREAD

**Issue by anchann** [fable-compiler/Fable#1975]
### Description

I have a fable app that I'm trying to run though webpack. It works fine, i.e. fable-loader logs `fable: Compiled $filename` for a few dozen files, and then I can hit `localhost:8080` and see my app. Then I reference an additional `.fsproj`. I re-run webpack, and the process consistently stalls at the `fable: Compiled alwaysTheSameFilename.fs`. There's a `dotnet` process taking 99% CPU at this point, which dies when I `ctrl+c` webpack.

### Repro code

I've spent an hour trimming down the referenced code, rearranging modules/namespaces, removing AutoOpens, and generally trying to narrow down problem. I'm done to having just three .fs files with one module per file, with mostly just type definitions in them, yet the problem persists.

I tried to find a way to turn up verbosity of fable or fable-loader, but couldn't find anything. Knowing which file it's stuck on would be a nice foothold for debugging this further.

Is there a way to get extra insight into what's going on inside fable?

### Related information

* Fable version: 2.4.13
* Operating system: MacOS 10.13.6

**Comment by anchann** [comment]
I've narrowed this down to the following repro case:

```
// Test.fs
[<AutoOpen>]
module Test

type TestType<[<Measure>] 't> =
    private | TestType of float<'t>

    member this.Value =
        match this with
        | TestType value -> value

let makeTestType (value: float<_>) : TestType<_> =
    TestType value


// App.fs
module App

open FSharp.Data.UnitSystems.SI.UnitSymbols

let test: TestType<m> = makeTestType 12.0

Fable.Core.JS.console.log "test"
Fable.Core.JS.console.log (test.Value)

App.fsproj
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <Compile Include="Test.fs" />
    <Compile Include="App.fs" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Fable.Core" Version="3.0.0" />
    <PackageReference Include="Fable.Elmish.React" Version="3.0.1" />
  </ItemGroup>
</Project>
```

Running this makes it hang at 99% CPU

```
$ npm run start

> @ start /Users/.../Workspace/github/fable2-samples/minimal
> webpack-dev-server

ℹ ｢wds｣: Project is running at http://localhost:8080/
ℹ ｢wds｣: webpack output is served from /
ℹ ｢wds｣: Content not from webpack is served from ./public
fable-compiler 2.3.19
fable: Compiled src/App.fsproj
```

Interestingly, making and using the value directly, without explicitly typing it, does not hang:

```
Fable.Core.JS.console.log ((makeTestType 12.0).Value)
```

Simply referencing the type without using it hangs:

```
module App

open FSharp.Data.UnitSystems.SI.UnitSymbols

type Blah = {
	  Field: TestType<m>
}

Fable.Core.JS.console.log "test"
```

**Comment by inosik** [comment]
I can reproduce in the REPL. However, the build-in types *which aren't aliases or combined units* as well as custom units of measure do work. For example: [`metre`](https://github.com/dotnet/fsharp/blob/fead0aac540485683f694524eadad79983ec28d9/src/fsharp/FSharp.Core/SI.fs#L9) does work, while [`meter`](https://github.com/dotnet/fsharp/blob/fead0aac540485683f694524eadad79983ec28d9/src/fsharp/FSharp.Core/SI.fs#L13) doesn't.

**Comment by inosik** [comment]
We run into an endless loop here:

https://github.com/fable-compiler/Fable/blob/b8b8594187b153838d385cbb290e6908c5195caf/src/Fable.Transforms/FSharp2Fable.Util.fs#L51-L59

`t.AbbreviatedType` is the same as `t`.

**Comment by MangelMaxime** [comment]
I am still new to FCS and I am writting this message without testing anything (that's bad I know :p)

But because F# measure are erased type I guess that's normal that `t` and `t.AbbreviatedType` are the same. Perhaps, we need to check for `IsMeasure` and make a special decision for them somewhere.

**Comment by anchann** [comment]
Thank you for fixing!

What's the process for getting this code released? I checked https://github.com/fable-compiler/Fable/releases and https://www.npmjs.com/package/fable-compiler. The released versions listed are oddly not the same (the latter has more), but the latest listed in either list is from 8 days ago, so probably doesn't have this fix in it.

**Comment by MangelMaxime** [comment]
Hello @anchann it should be released in 2.4.14.

The process for getting the code release is that a maintainer like me or @alfonsogarciacaro release the version. I am slowly taking over Fable maintenance and I still taken a bit of time to release new versions trying to not mess up the release :)

**Comment by anchann** [comment]
@MangelMaxime many thanks, for the release, the explanation of the process, and your previous and ongoing work on Fable! I've tried my code against the new version, and it doesn't get stuck anymore.
